File: combo-gen.py
import random
import string
import json
import time
import datetime
import sys
from faker import Faker

fake = Faker()

linestogenerate = sys.argv.pop() if len(sys.argv) > 1 else ""
if linestogenerate == "":
    print("No amount of lines to generate specified, defaulting to 50000")
    linestogenerate = 50000

time1 = time.perf_counter()

lowercase = string.ascii_lowercase
digits = string.digits
password = string.ascii_letters + string.digits + "_" + "." + "_" + "."
email = string.ascii_letters
emailadditive = ["_", "."]

with open("data/emaildomains.json", "r", encoding="utf-8") as domainfile:
    data = domainfile.read()
emaildomains = json.loads(data)

with open("data/firstnames.json", "r", encoding="utf-8") as firstnamefile:
    data = firstnamefile.read()
firstnames = json.loads(data)

with open("data/lastnames.json", "r", encoding="utf-8") as lastnamefile:
    data = lastnamefile.read()
lastnames = json.loads(data)

with open("data/passwords.json", "r", encoding="utf-8") as passfile:
    data = passfile.read()
passwords = json.loads(data)

with open("data/usernames.json", "r", encoding="utf-8") as usernamefile:
    data = usernamefile.read()
usernames = json.loads(data)

combofile = open(
    "generatedcredentials"
    + datetime.datetime.now().strftime("%m-%d-%Y-%H-%M-%S")
    + ".txt",
    "a",
    encoding="utf-8",
)


def randomName(lower):
    x = random.randint(1, 4)
    if x == 1:
        name = random.choice(firstnames)
    elif x == 2:
        name = random.choice(lastnames)
    elif x == 3:
        name = random.choice(firstnames) + random.choice(lastnames)
    elif x == 4:
        name = fake.name()
    if lower:
        name.lower()
    return name.replace(" ", "")


def USER():
    rand = random.randint(1, 3)
    if rand == 1:
        tempuser = randomName(True)
    elif rand == 2:
        return fake.email()
    else:
        tempuser = random.choice(usernames)

    username = ""
    first = True
    additive = False

    for letter in tempuser:
        int = random.randint(1, 50)
        if first:
            rand = random.randint(1, 20)
            if rand == 1:
                username = letter.upper()
            else:
                username = letter
            first = False
        else:
            if int <= 2:
                username = username + random.choice(email)
            elif int >= 48 and not additive:
                username = username + random.choice(emailadditive) + letter
                additive = True
            else:
                username = username + letter
    return username + "@" + random.choice(emaildomains)


def PASS():
    rand = random.randint(1, 6)
    name = randomName(False)
    if rand == 1:
        return "".join(random.sample(email, random.randint(6, 15)))
    elif rand == 2:
        return name + "".join(random.sample(digits, random.randint(2, 7)))
    elif rand == 3:
        str = ""
        if random.randint(1, 2) == 1:
            name.lower()
        for letter in name:
            if random.randint(1, 2) == 1:
                str = str + random.choice(password)
            else:
                str = str + letter
        return str
    return random.choice(passwords)


def COMBO():
    line = USER() + ":" + PASS()
    combofile.write(line + "\n")


for i in range(int(linestogenerate)):
    COMBO()

time2 = time.perf_counter()
print(str(linestogenerate) + " lines generated in " + str(round((time2 - time1), 3)) + " seconds.")

File: combo-gen.go
package main

import (
	_ "embed"
	"encoding/json"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"strings"
	"sync"
	"time"
)

//go:embed data/emaildomains.json
var emaildomains []byte
var emailDomainList []string

//go:embed data/firstnames.json
var firstnames []byte
var firstNameList []string

//go:embed data/lastnames.json
var lastnames []byte
var lastNameList []string

//go:embed data/passwords.json
var passwords []byte
var passwordList []string

//go:embed data/usernames.json
var usernames []byte
var usernameList []string

var emailadditive = []string{".", "_", "-"}
var letterList = []string{"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "x", "y", "z"}
var digitList = []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"}
var passwordCharList = []string{"!", "@", "#", "$", "%", "^", "&", "*", "(", ")", "-", "_", "=", "+", "[", "]", "{", "}", ";", ":", "'", "\"", ",", "<", ".", ">", "/", "?", "~"}

func randomName(lower bool) string {
	n := rand.Intn(2)
	name := ""

	if n == 0 {
		name = firstNameList[rand.Intn(len(firstNameList))]
	} else if n == 1 {
		name = lastNameList[rand.Intn(len(lastNameList))]
	} else if n == 2 {
		name = firstNameList[rand.Intn(len(firstNameList))] + lastNameList[rand.Intn(len(lastNameList))]
	}

	if lower {
		name = strings.ToLower(name)
	}

	return name
}

func getUser() string {
	n := rand.Intn(2)
	name := ""

	if n == 0 {
		name = randomName(true)
	} else if n == 1 {
		name = usernameList[rand.Intn(len(usernameList))]
	}

	userName := ""
	isFirst := true
	for i := 0; i < len(name); i++ {
		rnd := rand.Intn(50)
		letter := string(name[i])
		if rnd <= 2 {
			userName = userName + letterList[rand.Intn(len(letterList))] + letter
		} else if rnd >= 48 && !isFirst {
			userName = userName + emailadditive[rand.Intn(len(emailadditive))] + letter
		} else {
			userName = userName + letter
		}
		isFirst = false
	}

	return userName + "@" + emailDomainList[rand.Intn(len(emailDomainList))]
}

func getPassword() string {
	rnd := rand.Intn(5)
	name := randomName(false)

	if rnd == 0 {
		pw := ""
		for i := 0; i < rand.Intn(11)+6; i++ {
			pw = pw + letterList[rand.Intn(len(letterList))]
		}
		return pw
	} else if rnd == 1 {
		pw := ""
		for i := 0; i < rand.Intn(11)+6; i++ {
			pw = pw + digitList[rand.Intn(len(digitList))]
		}
		return pw
	} else if rnd == 2 {
		pw := ""
		if rand.Intn(2) == 0 {
			pw = strings.ToLower(name)
		}

		str := ""
		for i := 0; i < len(pw); i++ {
			if rand.Intn(2) == 0 {
				if rand.Intn(6) == 0 {
					str = str + passwordCharList[rand.Intn(len(passwordCharList))]
				} else {
					str = str + letterList[rand.Intn(len(letterList))]
				}
			} else {
				str = str + string(pw[i])
			}
		}

		return str
	}

	return passwordList[rand.Intn(len(passwordList))]
}

func getCombo() string {
	return getUser() + ":" + getPassword()
}

func main() {
	starTime := time.Now()
	count := 0
	if len(os.Args) < 2 {
		fmt.Println("No count provided, using default count of 50000")
		count = 50000
	} else {
		c, err := strconv.Atoi(os.Args[1])
		if err != nil {
			fmt.Println("Invalid count provided, using default count of 50000")
			count = 50000
		}

		fmt.Println("Count provided: ", c)
		count = c
	}

	json.Unmarshal(emaildomains, &emailDomainList)
	json.Unmarshal(firstnames, &firstNameList)
	json.Unmarshal(lastnames, &lastNameList)
	json.Unmarshal(passwords, &passwordList)
	json.Unmarshal(usernames, &usernameList)

	if _, err := os.Stat("output"); os.IsNotExist(err) {
		os.Mkdir("output", 0755)
	}

	newFile, error := os.Create("output/combos-" + time.Now().Format("2006-01-02-15-04-05") + ".txt")
	if error != nil {
		fmt.Println("Error creating file")
		return
	}

	defer newFile.Close()

	var wg sync.WaitGroup
	wg.Add(300)
	generated := 0

	for i := 0; i < 300; i++ {
		go func() {
			threadBuffer := ""
			threadBufferCount := 0
			for count > generated {
				combo := getCombo()

				threadBuffer = threadBuffer + combo + "\n"

				threadBufferCount++
				if threadBufferCount >= 10000 {
					newFile.WriteString(threadBuffer)
					threadBuffer = ""
					threadBufferCount = 0
				}

				generated++
			}

			if threadBuffer != "" {
				newFile.WriteString(threadBuffer)
			}
			wg.Done()
		}()
	}
	wg.Wait()

	fmt.Println("Combos generated successfully")
	fmt.Printf("Time taken: %.2f seconds", time.Since(starTime).Seconds())
}
